src/checkout/sections/Summary/Summary.tsx

```diff
@@ -39,7 +40,12 @@ export const Summary: FC<SummaryProps> = ({
 	discount,
 }) => {
 	return (
-		<div className="flex h-fit w-full flex-col overflow-hidden rounded lg:overflow-visible">
+		<div
+			className={clsx(
+				"z-0 flex h-fit w-full flex-col overflow-hidden rounded lg:overflow-visible",
+				"before:bottom-0 before:left-auto before:right-0 before:top-0 before:z-[-1] before:w-1/2 before:border-l before:border-neutral-200 before:bg-neutral-50 before:content-[''] before:lg:fixed",
```

Comment:
We can avoid rendering the `before` pseudoelement at all before it's needed – `before:lg:content-['']`